app/assets/javascripts/angular/controllers/graph_ctrl.js

```diff
@@ -123,7 +123,8 @@ angular.module("Prometheus.controllers").controller('GraphCtrl',
       var rangeSeconds = Prometheus.Graph.parseDuration(scope.graph.range);
       // bigger denominator == smaller step == more data
       var step = Math.floor(rangeSeconds / scalingFactor)
-      refreshFn(rangeSeconds, step).then(function(data) {
+      var endTime = Math.floor($scope.graph.endTime / 1000);
```

Comment:
What is the reason the endTime calculation needs to be moved here? Not that I think it's a bad idea to not always use the global $scope in all methods (adds so many side effects), more for my code understanding of the change of this PR.